src/serial.rs

```diff
@@ -111,6 +116,97 @@ cfg_if! {
     }
 }
 
+/// Types for configuring a serial interface.
+pub mod config {
+    use crate::time::rate::{Baud, Extensions};
+
+    // Reexport stop bit enum from PAC. In case there is a breaking change,
+    // provide a compatible enum from this HAL.
+    pub use crate::pac::usart1::cr2::STOP_A as StopBits;
+
+    /// Parity generation and checking. If odd or even parity is selected, the
+    /// underlying USART will be configured to send/receive the parity bit in
+    /// addtion to the data bits.
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    #[derive(Clone, Copy, PartialEq)]
+    pub enum Parity {
+        /// No parity bit will be added/checked.
+        ParityNone,
+        /// The MSB transmitted/received will be generated/checked to have a
+        /// even number of bits set.
+        ParityEven,
+        /// The MSB transmitted/received will be generated/checked to have a
+        /// odd number of bits set.
+        ParityOdd,
+    }
+
+    /// Configuration struct for [`Serial`](super::Serial) providing all
+    /// communication-related / parameters.
+    #[derive(Clone, Copy, PartialEq)]
+    pub struct Config {
+        /// Serial interface baud rate
+        pub baudrate: Baud,
+        /// Whether and how to generate/check a parity bit
+        pub parity: Parity,
+        /// The number of stop bits to follow the last data bit or the parity
+        /// bit
+        pub stopbits: StopBits,
+    }
+
+    impl Config {
+        /// Sets the baudrate to `baudrate`.
+        pub fn baudrate(mut self, baudrate: impl Into<Baud>) -> Self {
+            self.baudrate = baudrate.into();
+            self
+        }
+        /// Sets the parity to [`Parity::ParityNone`].
+        pub fn parity_none(mut self) -> Self {
+            self.parity = Parity::ParityNone;
+            self
+        }
+
+        /// Sets the parity to [`Parity::ParityEven`].
+        pub fn parity_even(mut self) -> Self {
+            self.parity = Parity::ParityEven;
+            self
+        }
+
+        /// Sets the parity to [`Parity::ParityOdd`].
+        pub fn parity_odd(mut self) -> Self {
+            self.parity = Parity::ParityOdd;
+            self
+        }
```

Comment:
Even though this reads better, I would suggest 
```suggestion
        /// Sets the parity
        pub fn parity(mut self, parity: Parity) -> Self {
            self.parity =  parity;
            self
        }
```

as it is for `fn stopbits()`

This would better compliment the change I suggested above (making the struct `#[non_exhaustive]`)

Reply:
Done.